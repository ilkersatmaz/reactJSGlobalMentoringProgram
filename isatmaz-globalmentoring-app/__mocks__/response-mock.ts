export const movieResponse = {
	data: {
		id: 221122,
		title: 'first movie of the year',
		overview:
			'overview of the video for the m ',		
		genres: ['Action', 'Horror'],
		budget: 1000,
		poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
		release_date: '2023-01-01',
		revenue: 136907000,
		runtime: 134,
		tagline: "new and best",
		
		vote_average: 9.9,
		vote_count: 1145,
	},

	totalAmount: 1,
};
