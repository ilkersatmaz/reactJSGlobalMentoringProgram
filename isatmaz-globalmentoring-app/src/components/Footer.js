import React from "react";

class Footer extends React.PureComponent {
    render() {
        return (
            <div className="footer-div">
                <h2>This my footer</h2>
            </div>
        )
    }
  }

export default Footer;