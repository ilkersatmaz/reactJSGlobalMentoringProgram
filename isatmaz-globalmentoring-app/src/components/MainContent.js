import React, {Component} from "react";
import MovieBoard from "./movieComponents/MovieBoard";
import PageBanner from "./pageBanner/PageBanner";
import Header from "./headerComponent/Header";
import MovieBanner from "./movieBanner/MovieBanner";

class MainContent extends Component{    

    constructor(props) {
        super(props);
    
        this.state = {
          selectedMovie: undefined,
        };
      }
    
    onMovieClick = (movie) => {
        this.setState({ selectedMovie: movie });
    };
    
    render(){
        return (
            <div>
                <Header isMovieBannerOpen= {this.state.selectedMovie ? true : false}  />
                {this.state.selectedMovie ? <MovieBanner movieData={this.state.selectedMovie}/> : <PageBanner />}                
                <MovieBoard onMovieClick={this.onMovieClick}/>           
            </div>
        )
    }
}

export default MainContent;