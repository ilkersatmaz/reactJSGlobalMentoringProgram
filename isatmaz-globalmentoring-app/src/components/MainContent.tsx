import React, {useCallback, useState} from "react";
import MovieBoard from "./movieComponents/MovieBoard";
import PageBanner from "./pageBanner/PageBanner";
import Header from "./headerComponent/Header";
import MovieBanner from "./movieBanner/MovieBanner";

function MainContent (){    
    const [selectedMovie, setSelectedMovie] = useState(undefined);
    const onMovieClick = useCallback((movie : any) => {
        setSelectedMovie(movie);
    },[]);

    const onSearchClick = useCallback(() => {
        setSelectedMovie(undefined)
    },[]);

    return (
        <div>
            <Header isMovieBannerOpen= {selectedMovie ? true : false} onSearchClick={onSearchClick}  />
            {selectedMovie ? <MovieBanner movieData={selectedMovie}/> : <PageBanner />}                
            <MovieBoard onMovieClick={onMovieClick}/> 
        </div>
    )
}

export default MainContent;