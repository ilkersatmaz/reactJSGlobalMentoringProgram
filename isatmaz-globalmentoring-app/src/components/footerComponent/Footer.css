.footer-div{
    position: relative;
    background-color: #424242;
    height: 70px;
}
.footer-text{
    position: absolute;
    height: 24px;
    left: 44%;
    right: 43.42%;
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 900;
    font-size: 20px;
    line-height: 24px;
    color: #F65261;
}