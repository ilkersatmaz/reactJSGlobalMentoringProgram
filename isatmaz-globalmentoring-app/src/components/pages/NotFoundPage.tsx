import React from 'react';

function NotFoundPage (props:any) {
  
  return (
    <div>                
        Not found
    </div> 
  );
}

export default NotFoundPage