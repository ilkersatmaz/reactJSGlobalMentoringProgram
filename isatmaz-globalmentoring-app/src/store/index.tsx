import { configureStore } from '@reduxjs/toolkit';
import moviesReducer from './movies/moviesSlice';
import movieOperationReducer from '../store/movies/movieOperationSlice';
export const store = configureStore({
    reducer: {
        movies: moviesReducer,
        selectedMovie: movieOperationReducer
    },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;