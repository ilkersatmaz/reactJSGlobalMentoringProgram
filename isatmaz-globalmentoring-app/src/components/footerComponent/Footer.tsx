import React from "react";
import './Footer.css';

function Footer(){
    return <div className="footer-div">
        <p className="footer-text"><b>netflix</b>roulette</p>
    </div>
}

export default Footer;