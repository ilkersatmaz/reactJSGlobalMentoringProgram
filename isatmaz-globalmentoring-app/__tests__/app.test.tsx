import React, { FC } from 'react';
import '@testing-library/jest-dom';
import { createMemoryRouter, Navigate, Route, RouterProvider, Routes } from 'react-router-dom';
import { Provider, ProviderProps } from 'react-redux';
import { fireEvent, render, screen } from '@testing-library/react';

import { addNewMovie, deleteSelectedMovie, fetchSelectedMovie, initialState, movieOperationSlice } from '../src/store/movies/movieOperationSlice';
import { fetchMovies } from '../src/store/movies/moviesSlice';
import { movieResponse } from '../__mocks__/response-mock';
import { store } from '../src/store/index';

import NotFoundPage from '../src/components/pages/NotFoundPage';
import MainContent from '../src/components/MainContent';
import Footer from '../src/components/footerComponent/Footer';

const ReduxProvider: FC<ProviderProps> = ({ children, store }) => <Provider store={store}>{children}</Provider>;

function renderWithContext() {
	render(
		<ReduxProvider store={store}>
				 <Routes>
                        <Route path="*" element={<NotFoundPage />} />
                        <Route path="/" element={<Navigate to="/search" replace />} />
                        <Route path="/search" element={
                        <>
                            <MainContent />
                            <Footer/>
                        </>} />
                    </Routes>			
		</ReduxProvider>,
	);
}

describe('Redux store tests', () => {
	it('Should return initial state', () => {
		const mockReducer = movieOperationSlice.reducer;
		expect(
			mockReducer(undefined, {
				type: undefined,
			}),
		).toEqual(initialState);
	});

	it('Fetch all movies action', async () => {
		const expectedState = {
			...initialState,
			movies: movieResponse.data,
			moviesCount: movieResponse.totalAmount,
			moviesLoadingStatus: 'succeeded',
		};

		expect(store.getState().movies).toEqual(expectedState);
	});

	it('Delete movie action', async () => {
		const expectedState = {
			...initialState,
			movies: movieResponse.data,
			moviesCount: movieResponse.totalAmount,
			moviesLoadingStatus: 'succeeded',
			movieFormLoading: 'pending',
		};

		expect(store.getState().movies).toEqual(expectedState);
	});

	it('Add movie action', async () => {
		const expectedState = {
			...initialState,
			movies: movieResponse.data,
			moviesCount: movieResponse.totalAmount,
			moviesLoadingStatus: 'succeeded',
			movieFormLoading: 'pending',
		};

		expect(store.getState().movies).toEqual(expectedState);
	});
});

describe('Component tests', () => {
	it('Search bar snapshot', async () => {
		renderWithContext();
		const element = screen.getByTestId('search-panel');
		expect(element).toMatchSnapshot();
	});	
});
