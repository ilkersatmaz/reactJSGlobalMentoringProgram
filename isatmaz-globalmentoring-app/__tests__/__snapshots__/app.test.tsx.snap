// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests Search bar snapshot 1`] = `
<div
  data-testid="search-panel"
>
  <a
    class="sc-hHTYSt hBClyJ"
    href="/search"
  >
    <img
      alt="logo-netflix-roulette"
      src="undefinedassets/images/logo.svg"
    />
    <button
      class="sc-gKPRtg sc-ipEyDJ crhnbk cjUwbM"
    >
      + add movie
    </button>
  </a>
  <div
    class="sc-jrcTuL iBFwKc"
  >
    <h1
      class="sc-bcXHqe sc-pyfCe jwEIAo uqdrl"
    >
      Find your movie
    </h1>
    <div
      class="sc-kDvujY kupgi"
    >
      <input
        class="sc-dmctIk eolsIr"
        data-testid="search-input"
        id="search-input"
        placeholder="What do you want to find"
        value=""
      />
      <button
        class="sc-gKPRtg sc-iBYQkv crhnbk caTMnA"
        data-testid="search-button"
        id="search-button"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
